### PR TITLE
Egress Gateway: Remove deprecated CENP CRD

### PR DESCRIPTION
**Pull request by julianwiedmann**
For `v1.13` we want to drop the support for configuring EgressGW policies via CENP, and rely on `CiliumEgressGatewayPolicy` instead (introduced in `v1.12`).

We've been warning about this in the release notes and via `deprecationWarning` in the CENP CRD itself, so it shouldn't come as surprise to anyone.

```release-note
egressgw: drop support for CiliumEgressNATPolicy
```

### CODE REVIEW

**Comment by julianwiedmann** [comment]
/test-1.16-4.9

**Comment by julianwiedmann** [comment]
[`1.16-4.9` flaked with `Unable to download helm chart v1.11 from GitHub`]

**Comment by julianwiedmann** [comment]
`ci-aks` seems to be hitting https://github.com/cilium/cilium/issues/22162.

File: pkg/egressgateway/manager.go [hunk 1]
```diff
@@ -117,12 +117,12 @@ func (manager *Manager) OnAddEgressPolicy(config PolicyConfig) {
 	manager.Lock()
 	defer manager.Unlock()
 
-	logger := log.WithField(logfields.CiliumEgressNATPolicyName, config.id.Name)
+	logger := log.WithField(logfields.CiliumEgressGatewayPolicyName, config.id.Name)
 
 	if _, ok := manager.policyConfigs[config.id]; !ok {
-		logger.Debug("Added CiliumEgressNATPolicy")
+		logger.Debug("Added CiliumEgressGatewayTPolicy")
```
Comment (jibi) [inline pkg/egressgateway/manager.go]:
```suggestion
		logger.Debug("Added CiliumEgressGatewayPolicy")
```

**Review by jibi — CHANGES_REQUESTED** [review]
Thanks! Only a few more things to remove:

Since we don't support CENP anymore, users can't specify just an egress IP (i.e. they need to specify an `egressGateway` with node selector etc), so these can go away as well:

https://github.com/cilium/cilium/blob/2b7bbe3baabb90690cf9db6b70d71ef7af7f2017/pkg/egressgateway/policy.go#L63

https://github.com/cilium/cilium/blob/2b7bbe3baabb90690cf9db6b70d71ef7af7f2017/pkg/egressgateway/policy.go#L90-L97

nit: NAT -> Gateway: https://github.com/cilium/cilium/blob/2b7bbe3baabb90690cf9db6b70d71ef7af7f2017/pkg/egressgateway/policy.go#L56

lastly these should be renamed as well: https://github.com/cilium/cilium/tree/master/examples/kubernetes-egress-gateway (and update their references in https://github.com/cilium/cilium/blob/master/Documentation/network/egress-gateway.rst)

**Comment by jibi** [comment]
> `ci-aks` seems to be hitting #22162.

(we enabled it back just to debug it, but it's not actually required at the moment)

File: test/k8s/manifests.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ var (
 	DemoDaemonSet = helpers.Manifest{
 		Filename:        "demo_ds.yaml",
 		Alternate:       "demo_ds_local.yaml",
-		DaemonSetNames:  []string{"testds", "testclient", "testclient-2"},
```
Comment (jibi) [inline test/k8s/manifests.go:16]: :cry:

**Review by jibi — APPROVED** [review]
Looks good, thanks! :ship: :ship:

**Comment by julianwiedmann** [comment]
[also needed to drop the privileged tests for the old-style configuration without node selector]

**Review by christarazi — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

**Review by kaworu — APPROVED** [review]
Thanks @julianwiedmann, Helm changes LGTM.

File: Documentation/network/egress-gateway.rst [hunk 3]
```diff
@@ -406,7 +406,7 @@ from the mediabot pod to leave the cluster with the IP of the Egress Gateway nod
 
 .. code-block:: shell-session
 
-    $ kubectl apply -f egress-nat-policy-egress-gateway.yaml
+    $ kubectl apply -f egress-gateway-policy.yaml
```
Comment (joestringer) [inline Documentation/network/egress-gateway.rst:409]: While we're at it, could we replace the "Example policy" direct yaml with a `.. include` for this file from the examples directory? (Could be a follow-up PR, I don't mind).

File: pkg/egressgateway/manager_privileged_test.go [hunk 4]
```diff
@@ -159,84 +157,6 @@ func (k *EgressGatewayTestSuite) TestEgressGatewayManager(c *C) {
...
-	// Create a new policy
-	policy3 := newEgressPolicyConfigWithNodeSelector("policy-3", ep1Labels, destCIDR, nodeGroup1Selector, testInterface1)
-	egressGatewayManager.OnAddEgressPolicy(policy3)
-
-	assertEgressRules(c, []egressRule{})
-	assertIPRules(c, []ipRule{})
-
-	// Add back ep-1 endpoint which matches policy-3
-	egressGatewayManager.OnUpdateEndpoint(&ep1)
-
```
Comment (joestringer) [inline pkg/egressgateway/manager_privileged_test.go:239]: I didn't quite catch, but what about the change in CRD format means that this test code is no longer necessary?

**Review by joestringer — APPROVED** [review]
LGTM for my codeowners.

File: pkg/egressgateway/manager_privileged_test.go (hunk 4 above)
Comment (julianwiedmann) [reply pkg/egressgateway/manager_privileged_test.go:239]: CEGP mandates a NodeSelector to pick the Gateway node (while CENP implicitly picked the node based on the configured `EgressIP`). So with CENP gone, we don't need to/can't test a policy without NodeSelector.

File: Documentation/network/egress-gateway.rst (hunk 3 above)
Comment (julianwiedmann) [reply Documentation/network/egress-gateway.rst:409]: ack, will follow-up :+1:

**Comment by julianwiedmann** [comment]
/test-1.16-4.9

**Comment by julianwiedmann** [comment]
/test-1.26-net-next

**Pull request merged**